packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-target-group.ts

```diff
@@ -280,11 +280,17 @@ export class NetworkTargetGroup extends TargetGroupBase implements INetworkTarge
         `Must be one of [${NLB_PATH_HEALTH_CHECK_PROTOCOLS.join(', ')}]`,
       ].join(' '));
     }
-    if (healthCheck.timeout && healthCheck.timeout.toSeconds() !== NLB_HEALTH_CHECK_TIMEOUTS[healthCheck.protocol]) {
-      ret.push([
-        'Custom health check timeouts are not supported for Network Load Balancer health checks.',
-        `Expected ${NLB_HEALTH_CHECK_TIMEOUTS[healthCheck.protocol]} seconds for ${healthCheck.protocol}, got ${healthCheck.timeout.toSeconds()}`,
-      ].join(' '));
+
+    const lowHealthCheckTimeout = 2;
+    const highHealthCheckTimeout = 120;
+    if (healthCheck.timeout) {
+      const timeoutSeconds = healthCheck.timeout.toSeconds();
+      if (timeoutSeconds < lowHealthCheckTimeout || timeoutSeconds > highHealthCheckTimeout) {
+        ret.push(`Health check timeout '${timeoutSeconds}' not supported. Must be a number between ${lowHealthCheckTimeout} and ${highHealthCheckTimeout}.`);
+      }
+      if (healthCheck.interval && healthCheck.interval.toSeconds() < timeoutSeconds) {
```

Comment:
Can we double check this? The console experience doesn't allow a health check to be configured unless the interval is strictly larger than the timeout. I think this should be:

`if (healthCheck.interval && healthCheck.interval.toSeconds() <= timeoutSeconds)`

Reply:
I put `healthCheck.interval.toSeconds() < timeoutSeconds` because I confirmed that I can make Target groups which is Timeout = Interval.


However I checked the console and it's true and I think your suggestion (`healthCheck.interval.toSeconds() <= timeoutSeconds`) would be more natural.

![target-group-100](https://github.com/aws/aws-cdk/assets/38136327/f4d42752-387d-4b92-8e0c-bec892478fcb)


I will fix this.

Reply:
Sorry, I realized that the change `healthCheck.interval.toSeconds() <= timeoutSeconds)` could potentially be affect a customer who is using `Timeout = Interval`

For instance, if a customer is using `HealthCheckIntervalSeconds = 10`, for HTTPS, HealthCheckTimeoutSeconds would be 10 (default value). This is a case of `Timeout = Interval`.

Reply:
Good catch. I agree.